doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,258 @@
+.. [2] Jules Scholler (2019) "Image denoising using inpainting"
+       https://www.jscholler.com/2019-02-28-remove-dots/
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import plotly.io
+import plotly.express as px
+from scipy import sparse
+
+import skimage as ski
+
+
+#####################################################################
+# The dataset we are using in this example is an image sequence (a movie!) of
+# human in-vivo tissue. Specifically, it shows the *palisades of Vogt* of a
+# given cornea sample.
+
+#####################################################################
+# Load image data
+# ===============
+
+image_seq = ski.data.palisades_of_vogt()
+
+print(f'number of dimensions: {image_seq.ndim}')
+print(f'shape: {image_seq.shape}')
+print(f'dtype: {image_seq.dtype}')
+
+#####################################################################
+# The dataset is an image stack with 60 frames (time points) and 2 spatial
+# dimensions. Let us visualize 10 frames by sampling every six time points:
+# We can see some changes in illumination.
+# We take advantage of the ``animation_frame`` parameter in
+# Plotly's ``imshow`` function. As a side note, when the
+# ``binary_string`` parameter is set to ``True``, the image is
+# represented as grayscale.
+
+fig = px.imshow(
+    image_seq[::6, :, :],
+    animation_frame=0,
+    binary_string=True,
+    labels={'animation_frame': '6-step time point'},
+    title='Sample of in-vivo human cornea'
+)
+plotly.io.show(fig)
+
+#####################################################################
+# Aggregate over time
+# ===================
+# First, we want to detect those dirt spots where the data are lost. In
+# technical terms, we want to *segment* the dirt spots (for
+# all frames in the sequence). Unlike the actual data (signal), the dirt spots
+# do not move from one frame to the next; they are still. Therefore, we begin
+# by computing a time aggregate of the image sequence. We shall use the median
+# image to segment the dirt spots, the latter then standing out
+# with respect to the background (blurred signal).
+# Complementarily, to get a feel for the (moving) data, let us compute the
+# variance.
+
+image_med = np.median(image_seq, axis=0)
+image_var = np.var(image_seq, axis=0)
+
+assert image_var.shape == image_med.shape
+
+print(f'shape: {image_med.shape}')
+
+fig, ax = plt.subplots(ncols=2, figsize=(12, 6))
+
+ax[0].imshow(image_med, cmap='gray')
+ax[0].set_title('Image median over time')
+ax[1].imshow(image_var, cmap='gray')
+ax[1].set_title('Image variance over time')
+
+fig.tight_layout()
+
+#####################################################################
+# Use local thresholding
+# ======================
+# To segment the dirt spots, we use thresholding. The images we are working
+# with have uneven illumination, which causes variations in the (absolute)
+# intensities of the foreground and the background, from one region to another
```

Comment:
```suggestion
# intensities of the foreground and the background, and from one region to another
```
Missing "and" here?

Reply:
No, it wouldn't make sense with this additional 'and:' We need local thresholding precisely because of the [variations] *from one region to another one*. There is comma before only because the noun phrase [variations] (which happens to be "variations in the (absolute) intensities of the foreground and the background") is very long, so it helps the reader, but really (grammatically) it shouldn't even be there.

Reply:
Ah, yes I see it now. Maybe we can make it a bit less complex instead? How about:

> The images we are working with are unevenly illuminated, which causes variations in the (absolute) intensities of the foreground and the background. E.g. the average background intensity in one region may be different in another (distant) one.

Reply:
Ok, but I think we still want to get to the point (i.e., there are intensity variations from one region to another, i.e., there are spatial variations in intensities), not only provide an example (but providing an example is great and makes it a nice read!). Building up on your suggestion, how about:

> The images we are working with are unevenly illuminated, which causes spatial variations in the (absolute) intensities of the foreground and the background. For example, the average background intensity in one region may be different in another (distant) one.

Reply:
Sound good! :+1:

Reply:
Done: a59a11f8c5